core/trino-main/src/main/java/io/trino/execution/DropRoleTask.java

```diff
@@ -61,6 +61,9 @@ public ListenableFuture<Void> execute(
         Session session = stateMachine.getSession();
         Optional<String> catalog = processRoleCommandCatalog(metadata, session, statement, statement.getCatalog().map(Identifier::getValue));
         String role = statement.getName().getValue().toLowerCase(ENGLISH);
+        if (!metadata.roleExists(session, role, catalog) && statement.isExists()) {
```

Comment:
I'd move `statement.isExists` as first operand (it's constant so cheaper if false and doesn't resolve metadata call).

Reply:
Done.